components/UWAPlantMap.vue

```diff
@@ -89,31 +120,50 @@ export default {
   methods: {
     loadMarkers() {
       if (this.map && this.google) {
-        // Clear old markers
-        for (const marker of this.markerInstances) {
-          marker.setMap(null)
-        }
-        this.markerInstances = []
+        this.clearMarkers()        
         // Create new markers and store them
-        this.markers.forEach((marker, index) => {
+        plants.forEach((plant, index) => {
           // Plot all instances
-          marker.instances.forEach(instance => {
-            const icon = iconPaths.hasOwnProperty(marker.type)
-                ? iconPaths[marker.type]
-                : iconPaths.info
-            Object.keys(iconStyle).forEach(style => {
-              icon[style] = iconStyle[style]
-            })
-            const markerInst = new this.google.maps.Marker({
-              label: marker.name,
-              position: instance.location,
-              icon,
-              map: this.map
-            })
+          plant.instances.forEach(instance => {
+            const markerInst = this.createMarkerInstance(plant, instance)
+            this.addListenerToMarker(markerInst, plant)
             this.markerInstances.push(markerInst)
           })
         })
       }
+    },
+    clearMarkers() {
+      this.markerInstances.forEach(marker => {
+        marker.setMap(null)
+      })
+      this.markerInstances = []
+    },
+    createMarkerInstance(plant, instance) {
+      const icon = iconPaths.hasOwnProperty(plant.type)
+                  ? iconPaths[plant.type]
+                  : iconPaths.info
+      Object.keys(iconStyle).forEach(style => {
+        icon[style] = iconStyle[style]
+      })
+      return new this.google.maps.Marker({
+        label: plant.name,
+        position: instance.location,
+        icon,
+        map: this.map
+      })
+    },
+    addListenerToMarker(markerInstance, plant) {
+      markerInstance.addListener('click', () => {
+        this.plantInfo = {
```

Comment:
Shouldn't be assigning the default values like this as it may cause issues if some of the data isn't populated.  Probably use a ternary on the entire object